src/Momento.Sdk/Internal/ScsDataClient.cs

```diff
@@ -36,17 +34,24 @@ protected Metadata MetadataWithCache(string cacheName)
     }
     protected DateTime CalculateDeadline()
     {
-        return DateTime.UtcNow.AddMilliseconds(dataClientOperationTimeoutMilliseconds);
+        return DateTime.UtcNow.Add(dataClientOperationTimeout);
     }
 
     /// <summary>
-    /// Converts TTL in seconds to milliseconds. Defaults to <c>defaultTtlSeconds</c>.
+    /// Converts TTL in seconds to milliseconds. Defaults to <see cref="ScsDataClientBase.defaultTtl" />.
     /// </summary>
-    /// <param name="ttlSeconds">The TTL to convert. Defaults to defaultTtlSeconds</param>
-    /// <returns></returns>
-    protected uint TtlSecondsToMilliseconds(uint? ttlSeconds = null)
+    /// <remark>
+    /// Conversion to <see langword="ulong"/> is safe here:
+    /// (1) we already verified <paramref name="ttl"/> strictly positive, and
+    /// (2) we know <paramref name="ttl.TotalMilliseconds"/> is less than <see cref="Int64.MaxValue"/>
+    /// because <see cref="TimeSpan"/> counts number of ticks, 1ms = 10,000 ticks, and max number of
+    /// ticks is <see cref="Int64.MaxValue"/>.
+    /// </remark>
+    /// <param name="ttl">The TTL to convert. Defaults to defaultTtl</param>
+    /// <returns>Milliseconds representation of the TTL (if provided, else of <see cref="defaultTtl"/>)</returns>
+    protected ulong TtlToMilliseconds(TimeSpan? ttl = null)
     {
-        return (ttlSeconds ?? defaultTtlSeconds) * 1000;
+        return (ulong)((ttl ?? defaultTtl).TotalMilliseconds);
```

Comment:
Could you use `Convert.ToUint64()` here?  In some languages a cast just means "pretend like this block of memory was this other data type".  IDK how c# represents doubles vs longs in memory or what it does with a cast like this but I would be surprised if the memory representation of the double would yield the same value if interpreted as a long without any conversion, so it seems like the `Convert` call would be safer/clearer.

Reply:
I'll add the statement. When jumping between languages the conventions vary so I agree it's clearer.

(Just leaving this as some documentation) .NET  does an actual conversion though, [see here](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions#explicit-conversions). To contrast going from `uint` to `int` (where you could have a positive integer outside the range of `int`) requires [extra validation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/checked-and-unchecked).